gui/mozregui/report_delegate.py

```diff
@@ -0,0 +1,51 @@
+from PyQt4.QtGui import QStyledItemDelegate, QStyleOptionProgressBarV2, \
+    QApplication, QStyle
+from PyQt4.QtCore import Qt, QRect, pyqtSlot as Slot
+
+
+class ReportItemDelegate(QStyledItemDelegate):
+    def __init__(self, parent=None, *args):
+        QStyledItemDelegate.__init__(self, parent, *args)
+        self.progress = 0
+        self.lastIndex = None
+
+    @Slot(object, int, int)
+    def show_dl_progress(self, dl, current, total):
```

Comment:
I feel like this logic should not be in the delegate, but in the model instead. That's the model job to keep data, and to signal that data has changed (model here is either the ReportModel object, or one of the ReportItem instance).